gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/DistributedIdentifyUtil.java

```diff
@@ -0,0 +1,309 @@
+                    if (!liveCommitPredicate.test(logEntry.getCommitMeta())) {
+                      // found first expired commit
+                      more = false;
+                    }
+                    // Process the commit if either the commit is considered live (i.e. the
+                    // liveCommitPredicate yields 'true') or at least for the very first commit
+                    // (aka the reference's HEAD).
+                    if (more || refHead) {
+                      action.accept(logEntry);
+                      refHead = false;
+                    }
+                  });
+          return more;
+        }
+      }.forEachRemaining(logEntry -> handleLiveCommit(logEntry, deletedKeys, bloomFilterMap));
+    } catch (NessieNotFoundException e) {
+      LOGGER.info("Reference {} to retrieve commits for no longer exists", reference);
+    }
+    return bloomFilterMap;
+  }
+
+  private IdentifiedResult walkAllReference(
+      NessieApiV1 api,
+      Reference reference,
+      Map<String, ContentBloomFilter> liveContentsBloomFilterMap) {
+    IdentifiedResult result = new IdentifiedResult();
+    try (Stream<LogResponse.LogEntry> commits =
+        StreamingUtil.getCommitLogStream(
+            api, "DETACHED", reference.getHash(), null, null, OptionalInt.empty(), true)) {
+      commits.forEach(logEntry -> handleCommit(logEntry, liveContentsBloomFilterMap, result));
+    } catch (NessieNotFoundException e) {
+      LOGGER.info("Reference {} to retrieve commits no longer exists", reference);
+    }
+    return result;
+  }
+
+  private void handleLiveCommit(
+      LogResponse.LogEntry logEntry,
+      Set<ContentKey> deletedKeys,
+      Map<String, ContentBloomFilter> bloomFilterMap) {
+    if (logEntry.getOperations() != null) {
+      logEntry
+          .getOperations()
+          .forEach(
+              operation -> {
+                if (operation instanceof Operation.Delete) {
+                  // collect the deleted keys
+                  deletedKeys.add(operation.getKey());
+                } else if (operation instanceof Operation.Put) {
+                  Content content = ((Operation.Put) operation).getContent();
+                  boolean isDropped = deletedKeys.contains(operation.getKey());
+                  if (!isDropped || bloomFilterMap.containsKey(content.getId())) {
+                    // If the content key is Not dropped, it is live commit.
+                    // If the content key is dropped and previously traversed commits have this
+                    // content id means
+                    // it is not a drop table, it is re-name table operation. So, consider it as
+                    // live.
+                    bloomFilterMap
+                        .computeIfAbsent(content.getId(), k -> new ContentBloomFilter(config))
+                        .put(content);
+                  }
+                }
+              });
+    }
+  }
+
+  private void handleCommit(
```

Comment:
Should the method name be "handleCommitForExpiredContents"  or so? same applies to it's call site